### PR TITLE
refactor: replaced layout context to zustand layout store

### PR DESCRIPTION
**Pull request by void-hr**
This PR resolves issue #31

- [x] removed layout.ts from hooks
- [x] changed are made only for zustand layout store

Let me know, if this will work or I'll ammend changes as mentioned

### CODE REVIEW

**Review by Daniel-Boll — APPROVED** [review]
It LGTM, I'll clone and run to test it out in a minute. Then I'll approve.

**Review by DanielHe4rt — APPROVED** [review]

File: src/lib/menu-list.ts
Comment (void-hr) [reply src/lib/menu-list.ts:39]: Looking forward to actively contribute. Actually I do not have clear idea about the web app or our scylla studio yet. 

I'll try to figure out what's going wrong 👀

**Comment by Daniel-Boll** [comment]
After pruning my Docker images and following the steps in the README again, I managed to run it successfully. What issues are you encountering?

**Comment by void-hr** [comment]
Just trying this command to run scylla studio via docker
```bash
 docker run --name scylla --network ws-scylla -p "9042:9042" -d scylladb/scylla:6.1.2 \
  --overprovisioned 1 \
  --smp 1
  ```
  
  localhost:3000  shows `This site can't be reached`

**Comment by Daniel-Boll** [comment]
> Just trying this command to run scylla studio via docker
> 
> ```shell
>  docker run --name scylla --network ws-scylla -p "9042:9042" -d scylladb/scylla:6.1.2 \
>   --overprovisioned 1 \
>   --smp 1
> ```
> 
> localhost:3000 shows `This site can't be reached`

Ow, no. This docker command is to run a local ScyllaDB instance (which you also need). The Scylla Studio docker command itself may be found a little bellow in the README.

```
docker run --rm --name scylla-studio --network="host" ghcr.io/basementdevs/scylla-studio:latest
```

**Comment by void-hr** [comment]
> > Just trying this command to run scylla studio via docker
> > ```shell
> >  docker run --name scylla --network ws-scylla -p "9042:9042" -d scylladb/scylla:6.1.2 \
> >   --overprovisioned 1 \
> >   --smp 1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > localhost:3000 shows `This site can't be reached`
> 
> Oh, no. This docker command is to run a local ScyllaDB instance (which you also need). The Scylla Studio docker command itself may be found a little bellow in the README.
> 
> ```
> docker run --rm --name scylla-studio --network="host" ghcr.io/basementdevs/scylla-studio:latest
> ```

Sorry, I was pasting the later one the command that you have shared  but ended up pasting the above one (by mistake).

**Comment by void-hr** [comment]
Also I am on windows, also tried it on wsl but getting same res localhost refused to connect

**Comment by Daniel-Boll** [comment]
Try changing the command to something like this:

```
docker run --rm -p 3000:3000 --name scylla-studio --network="ws-scylla" ghcr.io/basementdevs/scylla-studio:latest
```

Directly connection to scylla's db may help, but you will have to add a connection based on the IP of the ws-scylla docker network to the container of scylladb.

```
docker network inspect ws-scylla | jq ".[0].Containers"
```

find the scylla container and get it's ip e.g. 172.19.0.2 then you call add a connection to the DB using something like: `172.19.0.2:9042`  

This is a solution I think may work, but it's pretty much a workaround the default way as I cannot understand why it's not working in your case.

**Comment by void-hr** [comment]
> Also I am on windows, also tried it on wsl but getting same res localhost refused to connect

Update : I mapped port manually in the command itself and now it is working fine! 

`docker run --rm --name scylla-studio -p 3000:3000  ghcr.io/basementdevs/scylla-studio:latest`


> docker network inspect ws-scylla | jq ".[0].Containers"

If i only talk about localhost:3000 manually mapping port in existing command is working fine! Thanks a lot

**Comment by void-hr** [comment]
@Daniel-Boll thanks, this look cool

image

File: src/lib/menu-list.ts
Comment (void-hr) [reply src/lib/menu-list.ts:39]: @DanielHe4rt, continuing to the expected behaviour basically it should show list of all connection in menu select ?

Like this? 
image

Comment (DanielHe4rt) [reply src/lib/menu-list.ts:39]: Just like that!

Comment (void-hr) [reply src/lib/menu-list.ts:39]: Just added the following line in menu-list, however i have not called fetchInitalConnection on mount but still it is working. In both cases the data gets updated on reloading/refreshing  page 
```js
 const fetchInitialConnections = useLayout(
      (state) => state.fetchInitialConnections, 
    );
  ```

Comment (void-hr) [reply src/lib/menu-list.ts:39]: can you confirm this feature is working after adding connection followed by page refresh?

Comment (DanielHe4rt) [reply src/lib/menu-list.ts:39]: @Daniel-Boll could you please check it?

Comment (Daniel-Boll) [reply src/lib/menu-list.ts:39]: I was in high demands. I will try to give it a look tomorrow guys.

**Comment by DanielHe4rt** [comment]
@void-hr Did you pushed the latest changes?

**Comment by void-hr** [comment]
> @void-hr Did you pushed the latest changes?

No, I was waiting for confirmation if on hard refresh it is showing all connection in menu lists, so no need to push latest changes since it is of no use. Actually  menu list should update without hard refresh when adding connection but I am not able to figure out how we can achieve that. Used useEffect also with dependencies like connections and all but it is not working as expected. 

I am not sure but maybe we have to update the state on adding connection in `modal.tsx`

```js
const handleSave = (data: z.infer<typeof formSchema>) => {
    startTransition(async () => {
      await onSave(data);
      setOpen(false);
    });
  };
  ```

**Comment by DanielHe4rt** [comment]
@void-hr push a wip so we can help you with that. Maybe @Matozinho or @Daniel-Boll could help you with that.

**Comment by void-hr** [comment]
> @void-hr push a wip so we can help you with that. Maybe @Matozinho or @Daniel-Boll could help you with that.

Sure!

File: src/lib/menu-list.ts
Comment (DanielHe4rt) [reply src/lib/menu-list.ts:39]: I'll give it a look now.

Comment (DanielHe4rt) [reply src/lib/menu-list.ts:39]: Sorry by the delay on the feedback. So: 

Now that you probably have a ScyllaDB Database running in your local environment, would be a good thing if you check-out and see the default behavior. 

What we expect is that the connection selected triggers the Database SubMenus loading, not pop the update modal.

**Comment by Matozinho** [comment]
Hey @void-hr. I tested the code here and I found some issues.
- The `NewConnectionModal` is opening in weird moments
- The `New Connection` button opens the modal with the data to edit, so we couldn't add a new connection.
- The keyspaces are not fetched due to the `queryKeyspace` being called nowhere

https://github.com/user-attachments/assets/a731056b-fc3b-483a-8f99-3f1e228b4818

And I was running at the latest commit on your main:
image

File: src/contexts/layout.tsx
Comment (Matozinho) [inline src/contexts/layout.tsx:77]: The problem of removing the `useEffect` is that the keyspaces are not fetched anymore. Neither the initial connections (the `useEffect` above)

**Review by Matozinho — CHANGES_REQUESTED** [review]

**Comment by void-hr** [comment]
Noted! Work is in progress, trying to debug. Also can you please confirm that adding console log statements are working or not.

**Comment by Matozinho** [comment]
> Noted! Work is in progress, trying to debug. Also can you please confirm that adding console log statements are working or not.

They are, but depending on the file they will appear on the terminal, not the browser.

File: src/contexts/layout.tsx
Comment (void-hr) [reply src/contexts/layout.tsx:77]: @Matozinho the problem I was facing with hooks was that since it is not a component and is zustand store we can't use hook.

Comment (void-hr) [reply src/contexts/layout.tsx:77]: Sure will let you know!

**Comment by void-hr** [comment]
@Matozinho can you take a look again?

**Comment by DanielHe4rt** [comment]
> @Matozinho can you take a look again?

We will take a look on that today after our working hours. Thanks again for keep the contributions!

Btw, we added an Discord Channel recently. If you have any interest, feel free to join us: https://discord.gg/ShwwggRU

**Review by DanielHe4rt — APPROVED** [review]
I just fully tested it and seems to work flawlessly! Thanks @void-hr for the contribution.

In order to merge, I'll need a code review from one of the FE devs to make sure that has nothing missing.

**Review by Daniel-Boll — APPROVED** [review]

**Comment by Daniel-Boll** [comment]
Code-wise is looking good, as @DanielHe4rt already tested it LGTM to be merged

**Review by gvieira18 — APPROVED** [review]

**Comment by Daniel-Boll** [comment]
Thank you for your contribution @void-hr, I'm sorry for the time it took and the lack of support by my part which was the responsible for the issue.

I'd like to extend my gratitude for the team @Matozinho @gvieira18 @DanielHe4rt @diegoreis42 for taking care of the issue and being able to help everyone, thanks guys.

**Comment by void-hr** [comment]
@Daniel-Boll it was a nice experience, main reason was different timezone i guess. I thankyou all  of you  (@Matozinho @gvieira18 @DanielHe4rt @diegoreis42 @Daniel-Boll ) to guide me through!

**Pull request merged**
